swarm/api/services/run_state.py

```
@@ -174,12 +174,24 @@ def list_runs(self, limit: int = 20) -> List[Dict[str, Any]]:
+
+            # Check cache first
+            if run_id in self._cache:
+                state = self._cache[run_id]
+            else:
+                state_path = run_dir / "run_state.json"
+                if not state_path.exists():
+                    continue
+
+                try:
+                    state = json.loads(state_path.read_text(encoding="utf-8"))
+                except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)